tests/publiccloud/check_registercloudguest.pm

```diff
@@ -148,6 +149,50 @@ sub check_instance_unregistered {
     }
 }
 
+sub new_registration {
+    my ($instance) = @_;
+    record_info('Starting registration...');
+    $instance->ssh_script_retry(cmd => "sudo ${path}registercloudguest $regcode_param", timeout => 300, retry => 3, delay => 120);
+    check_instance_registered($instance);
+    return 0;
+}
+
+sub test_container_runtimes {
+    my ($instance) = @_;
+    my $image = "registry.suse.com/bci/bci-base:latest";
+
+    record_info('Install docker');
+    $instance->ssh_script_output("sudo rm ~/.docker/config.json");
+    #$instance->ssh_assert_script_run("source /etc/profile.local");
+    $instance->ssh_script_output("sudo zypper install -y docker");
+    record_info('Test docker');
+    $instance->ssh_script_output("sudo systemctl start docker.service");
+    record_info("systemctl status docker.service", script_output("systemctl status docker.service"));
```

Comment:
while from code wise  I would fully agree with you , test wise I must disagree.  Yes this will be code duplication but this code duplication has reasons . This test is to cover https://github.com/SUSE-Enceladus/cloud-regionsrv-client/blob/master/integration_test-process.txt which Public Cloud team is using to approve that certain package version of  `cloud-regionsrv-client` and we should not do less but also should not do more than defined in this document . On other hand `containers::common::install_docker_when_needed()` is our code which we using to prepare host for testing docker . For example it is already trying to activate containers module and restarting firewalld and NOT removing `~/.docker/config.json` which we need to do

Reply:
CORRECTION: - I wrote `while from code wise I would fully agree with you` before I checked actual code of `install_docker_when_needed()` . Now I change my mind , yes `install_docker_when_needed()` also installing docker but anyway it is totally different function with different purpose which not doing everything what you have deleted and also doing some things on top which under the question if they actually needed here or maybe they will even cause some regression :thinking: